clothcalc_placeholder.js

```diff
@@ -8274,6 +8277,106 @@
                     }
                 };
 
+                // ====================================================================
+                // april joke german
+                // ====================================================================
+                _self.injectDontTellAnyone = function () {
+                    var key = 'twdb_' + Character.playerId + '_april';
+                    var getLoc = function () {
+                        return JSON.parse(localStorage.getItem(key));
+                    };
+                    try {
+                        OptionsButler.twdb_activateRedeem = OptionsButler.twdb_activateRedeem || OptionsButler.activateRedeem;
+                        OptionsButler.activateRedeem = function (code) {
+                            if (code == 'lo0fl1rPa' && !localStorage.getItem(key)) {
+                                new west.gui.Dialog('Bonuscode benutzt', 'Schau in deinen Berichten nach.', 'ok').addButton('ok').show();
+                                Ajax.remoteCall('reports', 'get_reports', {}, function (json) {
+                                    localStorage.setItem(key, JSON.stringify([{
+                                        report_id:'01042017',
+                                        title:'Bonuscode eingelöst: lo0fl1rPa',
+                                        date:new ServerDate().date,
+                                        read:false,
+                                        publish_mode:0,
+                                        data_id:json.reports[0].data_id+1,
```

Comment:
I got an error on this line when running on public beta

Uncaught TypeError: Cannot read property 'data_id' of undefined

Looking at the code I assume this joke isn't meant for markets other than German... Then maybe the code should make sure it can't execute on other markets to minimize bugs on markets, where you have not tested it.